validator/evaluation/eval.py

```diff
@@ -215,36 +215,55 @@ def main():
 
     results_dict = {}
     for repo in lora_repos:
-        finetuned_model = PeftModel.from_pretrained(base_model, repo, is_trainable=False)
-        finetuned_model.eval()
-
         try:
-            is_finetune = model_is_a_finetune(original_model, finetuned_model)
+            # First try loading as a LoRA model
+            try:
+                finetuned_model = PeftModel.from_pretrained(base_model, repo, is_trainable=False)
+            except Exception as lora_error:
+                logger.info(f"Loading full model... failed to load as LoRA: {lora_error}")
+                # If LoRA loading fails, try loading as a full model
+                finetuned_model = AutoModelForCausalLM.from_pretrained(
+                    repo,
+                    token=os.environ.get("HUGGINGFACE_TOKEN")
+                )
+            finetuned_model.eval()
+
+            try:
+                is_finetune = model_is_a_finetune(original_model, finetuned_model)
```

Comment:
Do we need this if we are loading lora?  No right? Since we start with a base mode?

Reply:
True I can move it inside the except for the case where we load using `AutoModelForCausalLM.from_pretrained(repo...)`